src/ring/middleware/oauth2.clj

```diff
@@ -90,7 +90,15 @@
 (defn- parse-redirect-url [{:keys [redirect-uri]}]
   (.getPath (java.net.URI. redirect-uri)))
 
+(defn validate-profile
```

Comment:
Rather than packing an assert and a predicate together, you could write:

```clojure
(defn- valid-profile? [{:keys [client-id client-secret]}]
  (and (some? client-id) (some? client-secret)))
```

And then in `wrap-oauth2`, add a precondition:

```clojure
{:pre [(every? valid-profile? profiles)]}